src/api/utils.go

```diff
@@ -144,6 +144,27 @@ Gets the IP address of the client
 func GetClientAddr(r *http.Request) string {
 	// FIXME: check headers for X_HTTP_CLIENT_IP or something
 	// FIXME: error handling (third return value)
-	host, _, _ := net.SplitHostPort(r.RemoteAddr)
-	return host
+	srv := GetServer(r)
+	addr, _ := getProxiedClientAddr(srv, r)
+	return addr
+}
+
+func getProxiedClientAddr(srv *Server, r *http.Request) (string, error) {
+	srv_proxy := srv.proxy_addr
+	remote_addr, _, _ := net.SplitHostPort(r.RemoteAddr)
+	if real_ip, ok := r.Header["X_HTTP_REAL_IP"]; ok {
+		// Request was proxied
+		if srv_proxy == "" || srv_proxy == remote_addr {
```

Comment:
as in https://github.com/facebook/augmented-traffic-control/blob/master/atc/django-atc-api/atc_api/utils.py we should allow multiple proxies and default to 127.0.0.1 and ::1

This will allow using `X_HTTP_REAL_IP` header when srv_proxy is not set, which is not good.